default.nix

```diff
@@ -209,15 +209,16 @@ let
 
   # Stage 5: catch-all wrapper capable to run doom-emacs even
   # without installing ~/.emacs.d
-  # TODO: remove once Emacs 29+ is released and commonly available
   emacs = let
     load-config-from-site = writeTextDir "share/emacs/site-lisp/default.el" ''
       (message "doom-emacs is not placed in `doom-private-dir',
       loading from `site-lisp'")
-      ${lib.optionalString (!isEmacs29) ''
+      ${# TODO: remove once Emacs 29+ is released and commonly available
+        lib.optionalString (!isEmacs29) ''
         (load "${doom-emacs}/early-init.el")
-        (load "${doom-emacs}/core/core-start.el")
       ''}
+      (load "${doom-emacs}/lisp/doom.el")
+      (load "${doom-emacs}/lisp/doom-start.el")
```

Comment:
The entrypoint constantly changing makes me think we need to figure out some sort of API with upstream eventually.

Reply:
I am not sure how we could have an API here. The issue is that this is done very early during Emacs loading, so any kinda of API would fail unless we run Emacs with some special flag at build time?

Also, I think those changes should stop soon(-ish). doom-emacs is preparing they long awaited 3.0 release, and this is why it is having so many changes right now.